src/vulkan/buffer.cc

```diff
@@ -61,29 +63,11 @@ Result Buffer::CreateVkBufferView(VkFormat format) {
 }
 
 Result Buffer::CopyToDevice(VkCommandBuffer command) {
-  if (is_buffer_host_accessible_)
-    return FlushMemoryIfNeeded();
```

Comment:
Didn't this early exit before?

Reply:
We exist early here with this previous code. This is only the case where the buffer is **host-visible but not host-coherent**.
Now we allocate all buffer with only memory that is **host-visible and host-coherent**.
Now we can simply call memory barrier without exceptions.